lib/srv/regular/sshserver.go

```diff
@@ -860,19 +866,43 @@ func (s *Server) handleSessionRequests(sconn *ssh.ServerConn, identityContext sr
 	ctx.AddCloser(ch)
 	defer ctx.Close()
 
+	// Create a close context used to signal between the server and the
+	// keep-alive loop when to close th connection (from either side).
+	closeContext, closeCancel := context.WithCancel(context.Background())
+	defer closeCancel()
+
+	clusterConfig, err := s.GetAccessPoint().GetClusterConfig()
+	if err != nil {
+		log.Errorf("Unable to fetch cluster config: %v.", err)
+		return
+	}
+
+	// The keep-alive loop will keep pinging the remote server and after it has
+	// missed a certain number of keep-alive requests it will cancel the
+	// closeContext which signals the server to shutdown.
+	go srv.StartKeepAliveLoop(srv.KeepAliveParams{
```

Comment:
why don't you pass a context to keepalive loop as well?

Reply:
I am passing a context to the keep-alive loop?